package_library/packages_windows.go

```diff
@@ -182,6 +169,162 @@ func wuaUpdates(query string) ([]PkgInfo, error) {
 	return updates, nil
 }
 
+// InstallWUAUpdates installs all WUA updates that match query.
+func InstallWUAUpdates(query string) error {
+	connection := &ole.Connection{Object: nil}
+	if err := connection.Initialize(); err != nil {
+		return err
+	}
+	defer connection.Uninitialize()
+
+	unknown, err := oleutil.CreateObject("Microsoft.Update.Session")
+	if err != nil {
+		return err
+	}
+	defer unknown.Release()
+
+	session, err := unknown.QueryInterface(ole.IID_IDispatch)
+	if err != nil {
+		return err
+	}
+	defer session.Release()
+
+	updtsRaw, err := GetWUAUpdateCollection(session, query)
+	if err != nil {
+		return fmt.Errorf("GetWUAUpdates error: %v", err)
+	}
+	defer updtsRaw.Clear()
+
+	updts := updtsRaw.ToIDispatch()
+	count, err := updts.GetProperty("Count")
+	if err != nil {
+		return err
+	}
+	defer count.Clear()
+
+	updtCnt, _ := count.Value().(int32)
+
+	if updtCnt == 0 {
+		logger.Infof("No updates to install")
+		return nil
+	}
+
+	logger.Infof("%d Windows updates available", updtCnt)
+
+	var msg string
+	for i := 0; i < int(updtCnt); i++ {
+		if err := func() error {
```

Comment:
Just curious: what does this immediately invoked anonymous function get you? Is it just the per-iteration `defer`als? If so, would it be cleaner to factor that into a separate function call?